refactor(internal): create `Main` class

axel7083 opened the pull request:
### What does this PR do?

Introducing the `CodeMain` class, inspired from VSCode architecture.

Following @benoitf advise, instead of moving all `main/src/index.ts` content directly to the CodeMain, moving them bit by bit.

This first PR, is moving the index.ts code being called outside a function (globally).

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

### What issues does this PR fix or reference?

Part of https://github.com/podman-desktop/podman-desktop/issues/10042

### How to test this PR?

<!-- Please explain steps to verify the functionality,
do not forget to provide unit/component tests -->

- [x] Tests are covering the bug fix or the new feature

benoitf reviewed (commented):
question: wondering why the class or files are name

CodeMain and code-main

should it be just main / Main ?
or PodmanDesktopMain

to me it makes feel it's VSCode (named code internally)

axel7083 commented:
> to me it makes feel it's VSCode (named code internally)

I did not thought this was linked, I thought Code, like coding, xD, renaming to main to avoid confusion 👍

benoitf reviewed (approved):
LGTM

wondering why we rename App to ElectronApp ?

axel7083 commented:
> wondering why we rename App to ElectronApp ?

I found it messy to have App, same for Event (next PR, that I renamed to ElectronEvent), there are a lot of things already named, so I changed it to clarify when reading :/
